### ISSUE TITLE
srcery theme

### ISSUE THREAD

**Issue by anarcat** [tobi-wan-kenobi/bumblebee-status#881]
### Feature Request

I see bumblebee already has a lot of fine themes, but my favorite one ([srcery](https://srcery-colors.github.io/)) isn't featured. Anyone aware of an existing one? How easy is it making a theme?

**Comment by SamTebbs33** [comment]
It's pretty easy to make a new theme. I found it simple to copy a similar-looking one and adjust the colours used.  [Here's](https://github.com/SamTebbs33/dotfiles/blob/master/bumblebee-status/.config/bumblebee-status/themes/jellybeans.json) an example of my fairly minimal attempt.

**Comment by anarcat** [comment]
so i got this so far:

```
{
    "icons": [ ],
    "colors": [
        {
            "white": "#D0BFA1",
            "bright_white": "#FCE8C3",
            "black": "#1C1B19",
            "black_l": "#918175",
            "red": "#EF2F27",
            "green_d": "#519F50",
            "green_l": "#98BC37",
            "yellow_d": "#FBB829",
            "yellow_l": "#FED06E",
            "blue_d": "#2C78BF",
            "blue_l": "#68A8E4",
            "magenta_d": "#E02C6D",
            "magenta_l": "#FF5C8F",
            "cyan_d": "#0AAEB3",
            "cyan_l": "#53FDE9",
            "orange": "#D75F00",
            "bright_orange": "#FF8700",
            "xgray1": "#262626",
            "xgray2": "#303030",
            "xgray3": "#3A3A3A",
            "xgray4": "#444444",
            "xgray5": "#4E4E4E",
            "hard_black": "#121212"
        }
    ],
    "defaults": {
        "fg": "white",
        "bg": "black",
        "warning": {
            "bg": "orange",
            "fg": "hard_black"
        },
        "critical": {
            "bg": "red",
            "fg": "hard_black"
        }
    }
}
```

i'd love to add the awesome icons in there, but somehow they don't seem to load properly here, even if i have the fonts-awesome package installed in debian... does that ring a bell?

anyways, it feels like the two could be used seperately, no? like you pick the icons, the theme, separately?

**Comment by anarcat** [comment]
> anyways, it feels like the two could be used seperately, no? like you pick the icons, the theme, separately?

nevermind that, debian has font awesome 5, which this seems incompatible with. sigh.

**Comment by SamTebbs33** [comment]
Have you tried adding `"awesome-fonts"` to the font array?

**Comment by benthetechguy** [comment]
> debian has font awesome 5, which this seems incompatible with

From what I can see, Debian is on Font Awesome 4.7.0.

**Comment by tobi-wan-kenobi** [comment]
That is true, but *some* of the icons should still work, I think that indeed, the "iconset" setting is missing (but FontAwesome 5 is an issue, you are correct)

**Comment by anarcat** [comment]
> Have you tried adding `"awesome-fonts"` to the icons array?

yes, of course, that's how i found it "didn't work". :) here's a screenshot with the modified theme:

![snap-20220608T093643](https://user-images.githubusercontent.com/796623/172630406-c67cd339-f3ed-49fb-99fd-496c9990e9d5.png)

> From what I can see, Debian is on Font Awesome 4.7.0.

yeah my bad, it seems it was rolled back to fix precisely those kind of issues. i'm not sure what's going on then.

> _Some_ of the icons should still work, even with version 5. I think that indeed, the "iconset" setting is missing.

what is this refering to? the `icons` part of the theme? because it works *without* that (ie. i don't get garbled icons, but i don't get awesome icons either, there's just no icon).

i have also tried this:

        status_command bumblebee-status -m title pasink pasource cpu memory battery --iconset "awesome-fonts"

and that doesn't work either... it *does* show a little more icons though, it looks like this:

![snap-20220608T094121](https://user-images.githubusercontent.com/796623/172631392-605cb671-95a4-4a12-8208-4850ca651d9b.png)

so it looks like the default theme has separators that work correctly... but somehow they differ from what i get with my theme and `--iconset ascii` because I get bars, not arrows. not sure what i am doing wrong. :)

i also noticed the default theme has `urgent: true`... that's not documented anywhere I could find, what does it do? it seems important... if not urgent! ;)

thanks for all the replies!

**Comment by tobi-wan-kenobi** [comment]
Yes, with iconset, I meant the "icons" array in the theme JSON.

Note that the font you're using in the i3bar config *also* needs to have the required glyphs - which font are you using?

(sorry if that is an obvious question, but font config is a MAJOR pain, like 80% of all questions)

**Comment by anarcat** [comment]
On 2022-06-08 06:52:30, tobi-wan-kenobi wrote:
> Yes, with iconset, I meant the "icons" array in the theme JSON.

right, how does that interact with the --iconset parameter? can i keep
the "icons" blank and just let the user pick the icons they choose?

> Note that the font you're using in the i3bar config *also* needs to have the required glyphs - which font are you using?

oh. it looks like i had:

    font pango:DejaVu Sans Mono 10

but that's not actually what i wanted! So i set it to:

    font pango:Fira mono 10

... my goto UI font these days. ;)

it still doesn't fix the problem. should i add awesome in there?

i also tried the following, from the example in the README:

    bar {
        font pango:Inconsolata 10
        position top
        [...]

that doesn't work either... it changes the font, but the icons are still garbled.

> (sorry if that is an obvious question, but font config is a MAJOR pain, like 80% of all questions)

not at all, it's super hard, and not obvious to *me* anyways. :)

happy to do any further debugging you need here, and this seems to me
like stuff that could be added to a troubleshooting section eventually :)

-- 
La publicité est la dictature invisible de notre société.
                        - Jacques Ellul

**Comment by tobi-wan-kenobi** [comment]
Yes, adding font awesome as backup font should help (or only font awesome, for troubleshooting). The Inconsolata font only works if it is patches correctly, I think. It essentially boils down to the question "are the right glyphs in the font" and that - oh fun - differs from distro to distro. I have started just installing the powerline fonts from their girhub repo, si that fonts behave more predictably.

The iconset parameter on the cli overrides the theme setting, iirc.

Yeah, this should def. make it in a troubleshooting section.

Thanks a lot for your patience!
